data_hub_api/main.py

```diff
@@ -32,6 +32,16 @@ def get_root():
     def get_enhanced_preprints_docmaps_index():
         return enhanced_preprints_docmaps_provider.get_docmaps_index()
 
+    @app.get("/enhanced-preprints/docmaps/v1/get-by-doi")
```

Comment:
maybe it looks better without the `get-` and maybe, since the parameter is called that, it could be `by-preprint-doi`. Just a thought.